pkg/config/config.go

```diff
@@ -250,6 +250,15 @@ func (vm *VirtualMachine) AddIgnitionFileFromCmdLine(cmdlineOpts string) error {
 	return nil
 }
 
+func (vm *VirtualMachine) ValidateDevices() error {
+	for _, device := range vm.Devices {
+		if err := device.Validate(); err != nil {
```

Comment:
There are a number of calls to `Validate` in virtio.go, are they made redundant by this new method? or do we need both?

Reply:
I think either the Validate() calls at the end of `FromOptions` in some places can be removed or we can add dev.Validate to the end of all `FromOptions`. I have removed the redundant occurrences in this iteration.

Reply:
One concern I have with this is that this weakens the `FromOptions` and `AddDevicesFromCommandLine` APIs, which are kind of public: I expect other go programs to be using `github.com/crc-org/pkg/config`, even if I’m not sure these specific methods will be widely used.
Before this change, they were returning an error when the config does not validate, after this change, all users of the API need to remember to do the validation themselves or they could get invalid device configuration.

I would keep the validation at the end of each `FromOptions`, or add the validation to either `AddDevicesFromCmdLine` or `deviceFromCmdLine` if we want to avoid the duplication.

Reply:
Ah yes, that makes sense. I will add validate back to FromOptions